igvc_perception/src/pointcloud_segmentation/ground_filter.cpp

```diff
@@ -0,0 +1,48 @@
+#include "ground_filter.h"
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl_ros/point_cloud.h>
+
+typedef pcl::PointCloud<pcl::PointXYZ> PC;
+
+GroundFilterNode::GroundFilterNode()
+{
+  private_nh_ = ros::NodeHandle("~");
+  raw_pts_sub_ = private_nh_.subscribe("/lidar/transformed", 1, &GroundFilterNode::groundFilterCallback, this);
+  ground_filter_pub_ = private_nh_.advertise<sensor_msgs::PointCloud2>("ground_segmentation", 1);
+};
+
+void GroundFilterNode::groundFilterCallback(const sensor_msgs::PointCloud2ConstPtr& cloud_msg)
+{
+  PC::Ptr cloud(new PC);
+  PC::Ptr cloud_filtered(new PC);
+  pcl::fromROSMsg(*cloud_msg, *cloud);
+  const auto& x = 5.0;
+  const auto& y = 5.0;
+  const auto& z = 2.0;
```

Comment:
These values should be pulled from a configuration file instead of being hard coded